ADR/0038-integration-service-composite-removal.md

```diff
@@ -0,0 +1,31 @@
+# 38. Integration service removes composite snapshots and logic around them
+
+* Date started: 2024-07-10
+
+## Status
+
+
+## Context
+
+Composite snapshots main goal was to prevent race condition when teams merged multiple PRs to multiple components
+of the same application at nearly the same time. This concept was confusing for users and we managed to simplify it
+by immediate promotion to GCL using override snapshots. Users also ran into the issue with GCL deadlock. This was 
+caused by multiple components present in GCL causing tests for a new component fail if they were bundled into a snapshot.
+This deadlock prevented the relase of new version of the application.
```

Comment:
While I understand the GCL deadlock, I think that this part is confusing. 

Is it true that we would not reach deadlock if the required integration test scenarios _would have been passing_ on the GCL snapshot? We saw deadlock situations when users had an EC failure on multiple components in the GCL (i.e. the action of updating a single component in a composite snapshot would still result in a failure from a different component)?

Another situation that we could have encountered this problem is if there were two tightly coupled components where both needed to be tested together before it would be possible to get a passing ITS (i.e. if there were integration tests running).

Reply:
Yes that is correct, I tried to rephrase sentences about deadlock, let me know if that is making more sense now.